av/video/frame.pyx

```diff
@@ -282,10 +292,24 @@ cdef class VideoFrame(Frame):
         return frame
 
     @staticmethod
-    def from_ndarray(array, format='rgb24'):
+    def from_ndarray(array, format='rgb24', palette=None):
         """
         Construct a frame from a numpy array.
+
+        .. note:: ``palette`` must be given only for ``pal8``, and in ARGB format
+        (PyAV will swap bytes according to the endianness).
         """
+        if format == 'pal8':
+            assert array.dtype == 'uint8'
+            assert array.ndim == 2
+            assert palette.dtype == 'uint8'
+            assert palette.shape == (256, 4)
+            frame = VideoFrame(array.shape[1], array.shape[0], format)
+            copy_array_to_plane(array, frame.planes[0], 1)
+            frame.planes[1].update(palette.view('>i4').astype('i4').tobytes())
+            return frame
+        assert palette is None
```

Comment:
Can this be a `ValueError` with a message please?

Reply:
I've applied the change mentioned in https://github.com/mikeboers/PyAV/pull/601#issuecomment-589947756, which removes the need for an extra parameter and makes the API more consistent.